src/ast/expr.rs

```diff
@@ -486,6 +487,18 @@ impl Expr {
                 Dynamic::from_map(map)
             }
 
+            #[cfg(not(feature = "no_object"))]
+            #[cfg(feature = "indexmap")]
+            Self::Map(x, ..) if self.is_constant() => {
```

Comment:
Here and in eval.rs, the code use BTreeMap directly instead of crate::Map, I don't know if changing it would have ant side-effects, so I just add new code for indexmap feature.